From ab1f75e4a0a33f2f855a85c3eef7dfad1df555a9 Mon Sep 17 00:00:00 2001
From: Anish Trivedi <anish@freescale.com>
Date: Fri, 19 Mar 2010 10:54:04 -0500
Subject: [PATCH] ENGR00121798 MX53 RTC: Disable coin cell charging by pmic

Cleared COINCHEN bit in Power Control 0 register of MC13892.
Fixed spelling of #defines.

Signed-off-by: Anish Trivedi <anish@freescale.com>
---
 arch/arm/mach-mx5/mx51_3stack_pmic_mc13892.c  |    8 ++++----
 arch/arm/mach-mx5/mx51_babbage_pmic_mc13892.c |    8 ++++----
 arch/arm/mach-mx5/mx53_evk_pmic_mc13892.c     |   10 +++++-----
 3 files changed, 13 insertions(+), 13 deletions(-)

diff --git a/arch/arm/mach-mx5/mx51_3stack_pmic_mc13892.c b/arch/arm/mach-mx5/mx51_3stack_pmic_mc13892.c
index 6206b62..84c33ec 100644
--- a/arch/arm/mach-mx5/mx51_3stack_pmic_mc13892.c
+++ b/arch/arm/mach-mx5/mx51_3stack_pmic_mc13892.c
@@ -39,8 +39,8 @@
 #define STANDBYSECINV_WID 1
 
 /* Coin cell charger enable */
-#define CIONCHEN_LSH	23
-#define CIONCHEN_WID	1
+#define COINCHEN_LSH	23
+#define COINCHEN_WID	1
 /* Coin cell charger voltage setting */
 #define VCOIN_LSH	20
 #define VCOIN_WID	3
@@ -309,8 +309,8 @@ static int mc13892_regulator_init(struct mc13892 *mc13892)
 	pmic_write_reg(REG_POWER_CTL2, value, register_mask);
 
 	/* Enable coin cell charger */
-	value = BITFVAL(CIONCHEN, 1) | BITFVAL(VCOIN, VCOIN_3_0V);
-	register_mask = BITFMASK(CIONCHEN) | BITFMASK(VCOIN);
+	value = BITFVAL(COINCHEN, 1) | BITFVAL(VCOIN, VCOIN_3_0V);
+	register_mask = BITFMASK(COINCHEN) | BITFMASK(VCOIN);
 	pmic_write_reg(REG_POWER_CTL0, value, register_mask);
 
 	mc13892_register_regulator(mc13892, MC13892_SW1, &sw1_init);
diff --git a/arch/arm/mach-mx5/mx51_babbage_pmic_mc13892.c b/arch/arm/mach-mx5/mx51_babbage_pmic_mc13892.c
index cb442e4..4aa15f3 100644
--- a/arch/arm/mach-mx5/mx51_babbage_pmic_mc13892.c
+++ b/arch/arm/mach-mx5/mx51_babbage_pmic_mc13892.c
@@ -36,8 +36,8 @@
 #define uV_to_V(uV) (uV_to_mV(uV) / 1000)
 
 /* Coin cell charger enable */
-#define CIONCHEN_LSH	23
-#define CIONCHEN_WID	1
+#define COINCHEN_LSH	23
+#define COINCHEN_WID	1
 /* Coin cell charger voltage setting */
 #define VCOIN_LSH	20
 #define VCOIN_WID	3
@@ -363,8 +363,8 @@ static int mc13892_regulator_init(struct mc13892 *mc13892)
 	}
 
 	/* Enable coin cell charger */
-	value = BITFVAL(CIONCHEN, 1) | BITFVAL(VCOIN, VCOIN_3_0V);
-	register_mask = BITFMASK(CIONCHEN) | BITFMASK(VCOIN);
+	value = BITFVAL(COINCHEN, 1) | BITFVAL(VCOIN, VCOIN_3_0V);
+	register_mask = BITFMASK(COINCHEN) | BITFMASK(VCOIN);
 	pmic_write_reg(REG_POWER_CTL0, value, register_mask);
 
 #if defined(CONFIG_RTC_DRV_MXC_V2) || defined(CONFIG_RTC_DRV_MXC_V2_MODULE)
diff --git a/arch/arm/mach-mx5/mx53_evk_pmic_mc13892.c b/arch/arm/mach-mx5/mx53_evk_pmic_mc13892.c
index 805b3e5..681618b 100644
--- a/arch/arm/mach-mx5/mx53_evk_pmic_mc13892.c
+++ b/arch/arm/mach-mx5/mx53_evk_pmic_mc13892.c
@@ -39,8 +39,8 @@
 #define STANDBYSECINV_WID 1
 
 /* Coin cell charger enable */
-#define CIONCHEN_LSH	23
-#define CIONCHEN_WID	1
+#define COINCHEN_LSH	23
+#define COINCHEN_WID	1
 /* Coin cell charger voltage setting */
 #define VCOIN_LSH	20
 #define VCOIN_WID	3
@@ -307,9 +307,9 @@ static int mc13892_regulator_init(struct mc13892 *mc13892)
 	register_mask = BITFMASK(STANDBYSECINV);
 	pmic_write_reg(REG_POWER_CTL2, value, register_mask);
 
-	/* Enable coin cell charger */
-	value = BITFVAL(CIONCHEN, 1) | BITFVAL(VCOIN, VCOIN_3_0V);
-	register_mask = BITFMASK(CIONCHEN) | BITFMASK(VCOIN);
+	/* Disable coin cell charger since the cell is not rechargeable */
+	value = BITFVAL(COINCHEN, 0) | BITFVAL(VCOIN, VCOIN_3_0V);
+	register_mask = BITFMASK(COINCHEN) | BITFMASK(VCOIN);
 	pmic_write_reg(REG_POWER_CTL0, value, register_mask);
 
 	mc13892_register_regulator(mc13892, MC13892_SW1, &sw1_init);
-- 
1.5.4.4

